trl/trainer/sft_trainer.py

```diff
@@ -276,27 +276,43 @@ def _prepare_non_packed_dataloader(
         self, tokenizer, dataset, dataset_text_field, max_seq_len, formatting_func=None
     ):
         use_formatting_func = formatting_func is not None and dataset_text_field is None
+        self._dataset_sanity_checked = False
 
         # Inspired from: https://huggingface.co/learn/nlp-course/chapter7/6?fw=pt
         def tokenize(element):
+            input_batch = []
+            attention_masks = []
+
             outputs = tokenizer(
                 element[dataset_text_field] if not use_formatting_func else formatting_func(element),
                 truncation=True,
+                padding=True,
```

Comment:
@younesbelkada  this code is still incorrect - consider the case where **all** samples in the dataset are less than `max_seq_len`. Each batch will be padded to the largest element in the batch, but no data will pass the `if length == max_seq_len` check below.

Perhaps:
```
                padding='max_length',

Reply:
Yes you are correct thanks a lot for flagging, do you want to open a PR for that? happy to do it otherwise